homeassistant/components/device_tracker/legacy.py

```diff
@@ -115,6 +117,10 @@ def see(self, mac: str = None, dev_id: str = None, host_name: str = None,
 
         This method is a coroutine.
         """
+        if not self._registry_task:
```

Comment:
It's not necessary to cache this task, it's already cached inside `async_get_registry`.

Reply:
I need to wait somehow for the registry to be ready, early in the see method, to avoid race between consecutive service calls, where the first call waits for the registry before creating the device. Then the second call will have checked for existing device before the first call has created the device, spawning two device created events, when there should only be one.

Reply:
You mean I can just `await async_get_registry()`?

Reply:
Yes, that works. Thanks!